Fbf8'2017

mahikaw opened the pull request:
FBF8 sample JSON files to event here https://www.fbf8.com
Adresses #162 
This is the web app https://mahikaw.github.io/fbf8/

mahikaw commented:
@arp95 @niranjan94 @mariobehling Please review.

niranjan94 commented on sample/F8/event:
```diff
@@ -0,0 +1,49 @@
+    "end_time": "2017-04-19T17:00:00", 
+    "timezone": "US/Mountain", 
+    "description": "Join us for our annual 2-day event, where developers and businesses come together to explore the future of technology. Learn how Facebook connects the world through new products and innovation. This year’s event is bigger than ever – with more than 50 sessions, interactive experiences, and the opportunity to meet one-on-one with members of the Facebook team.", 
+    "background_image": "/images/background.png.png", 
```
Please rename the image to `background.png`

mahikaw commented:
@niranjan94 I need to add some images. So i'll squash them at the end. Do you see anything else that needs to be changed, in the JSON specifically?

niranjan94 commented on sample/F8/session_types:
```diff
@@ -0,0 +1,162 @@
+    {
+        "id": 22, 
+        "name": "Talk of 60 minutes", 
+        "length": "00:30"
```
TItles says 60 mins. length says 30 mins.

niranjan94 commented on sample/F8/session_types:
```diff
@@ -0,0 +1,162 @@
+    {
+        "id": 20, 
+        "name": "Workshop of 5 hours", 
+        "length": "00:30"
```
Fix length

niranjan94 commented on sample/F8/session_types:
```diff
@@ -0,0 +1,162 @@
+    {
+        "id": 19, 
+        "name": "Talk of 40 minutes", 
+        "length": "00:30"
```
Fix length

niranjan94 commented on sample/F8/session_types:
```diff
@@ -0,0 +1,162 @@
+    {
+        "id": 17, 
+        "name": "Workshop of 4 hours", 
+        "length": "00:30"
```
Fix length

niranjan94 commented on sample/F8/session_types:
```diff
@@ -0,0 +1,162 @@
+    {
+        "id": 16, 
+        "name": "Workshop of 1.5 hours", 
+        "length": "00:30"
```
Fix length

niranjan94 commented on sample/F8/session_types:
```diff
@@ -0,0 +1,162 @@
+    {
+        "id": 15, 
+        "name": "Workshop of 2.5 hours", 
+        "length": "00:30"
```
Fix length

niranjan94 commented on sample/F8/session_types:
```diff
@@ -0,0 +1,162 @@
+    {
+        "id": 14, 
+        "name": "Workshop of 1 hour", 
+        "length": "00:30"
```
Fix length

niranjan94 commented:
> Are the number of tracks and sessions more than enough? Or should I add more? 

I don't understand. More than enough ? You should add all the tracks and sessions of the FBF8 event.

mahikaw commented:
I mean like for Google I/O, arpit limited the speaker sample to 25 as there were too many speakers. Same is the case here. There are a lot of sessions, I have covered a significant amount though. Still, I could add those as well if you want.

niranjan94 commented:
@mahikaw oh meant like that. I'm not very sure about this. Let's ask @mariobehling .

arp95 commented on sample/F8/event:
```diff
@@ -0,0 +1,49 @@
+    "id": 180, 
+    "name": "F8-Facebook Developer Conference 2017", 
+    "latitude": 0.0, 
+    "longitude": 0.0, 
```
please add the latitude and longitude coordinates.

arp95 commented on sample/F8/microlocations:
```diff
@@ -0,0 +1,74 @@
+        "floor": null, 
+        "latitude": null, 
+        "longitude": null, 
+        "room": null
```
please add coordinates here as well.

arp95 commented on sample/F8/sessions:
```diff
@@ -0,0 +1,1249 @@
+            "id": 5,
+            "name": "Emil Sjölander"
+        }]
+    }, {
```
some session have tags while some not. please maintain your json structure

arp95 commented on sample/allofhands_hawaii2016/config.json:
```diff
@@ -0,0 +1 @@
+{"Api_Link": "https://raw.githubusercontent.com/mahikaw/open-event/hawaii/sample/allofhands_hawaii2016", "Email": "mahika.wason@gmail.com", "App_Name": "All Hands/2016 Hawaii"}
```
the api link is wrong. please check again.

Shailesh351 replied:
Why in this PR?

iamareebjamal replied:
Please change app name to FB F8

mahikaw commented:
@mariobehling @niranjan94 @arp95 @heysadboy Please review this sample.

Shailesh351 commented on sample/F8/speakers:
```diff
@@ -0,0 +1,2512 @@
+		"mobile": null,
+		"photo": null,
+		"organisation": "Facebook",
+		"sessions": [{
```
"sessions" array is at last in speaker object. It would be good to follow same pattern as generated by server.

Shailesh351 commented on sample/F8/speakers:
```diff
@@ -0,0 +1,2512 @@
+		"github": null,
+		"linkedin": null,
+		"position": "Product Manager"
+
```
Avoid unnecessary blank lines

Shailesh351 commented on sample/F8/speakers:
```diff
@@ -0,0 +1,2512 @@
+		"mobile": null,
+		"photo": null,
+		"organisation": "Facebook",
+		"sessions": [{
```
Same here. Maintain indentation of curly braes

Shailesh351 commented on sample/F8/tracks:
```diff
@@ -0,0 +1,209 @@
+		],
+		"track_image_url": null
+	}
+
```
Avoid unnecessary blank lines

Shailesh351 commented on sample/F8/speakers:
```diff
@@ -0,0 +1,2512 @@
+[{
```
Please maintain spacing and indentation i.e. curly braces of an object should be at same indentation.

It is creating problem on loading data from assets when latest data is not downloaded

Shailesh351 commented on sample/F8/tracks:
```diff
@@ -0,0 +1,209 @@
+[{
+		"id": 1,
+		"name": "ADS & MONETIZATION",
```
Track name is capital ?

iamareebjamal replied:
This is malformed JSON. Please pass the JSONs through a formatter and validator. The issue here is that `"` and `"` are not same.

Please refer this - https://jsonformatter.curiousconcept.com/

Paste the tracks json here

mahikaw replied:
@iamareebjamal I had already validated the JSON in jsonlint.com I tried your link too. It says that this is a valid JSON.

iamareebjamal replied:
![screenshot_20170605_154804](https://cloud.githubusercontent.com/assets/3874064/26780227/6d8c0888-4a06-11e7-9f91-b572ab09e1de.png)

iamareebjamal replied:
The GSON parser in the Andoid client says this too

iamareebjamal replied:
![screenshot_20170605_154934](https://cloud.githubusercontent.com/assets/3874064/26780278/9c5ca4d8-4a06-11e7-9127-8fa23626d4dc.png)

You can see the difference in quotes here

mahikaw replied:
This is weird. I don't say anything of that sorts on my screen. What we can do is that you could tell me the exact areas that it's giving the errors and I will re-write them?
<img width="597" alt="screen shot 2017-06-05 at 3 54 17 pm" src="https://cloud.githubusercontent.com/assets/21010060/26780392/478e3420-4a07-11e7-9b8e-dbea3cf563f0.png">

iamareebjamal replied:
See in raw format, it should be visible

Easiest thing will be to copy the quotation mark from `"name"` and paste it around `ADS & MONETIZATION` and `C`

mahikaw replied:
I just tried doing that. Git doesn't identify any change and so I can't even push it. @iamareebjamal

iamareebjamal replied:
I'll send a PR to your branch

mariobehling commented:
@mahikaw How can we avoid long pauses before a PR gets closed? It does not make sense to keep this open for such a long time. Please finish a PR before you do the next one. Please help to keep the development process flowing.

mahikaw commented:
@mariobehling I have incorporated the changes and it is good to merge from my side.
@arp95 @Shailesh351 @heysadboy @shreyasnbhat @iamareebjamal please give in your final reviews.

iamareebjamal commented on sample/F8/session_types:
```diff
@@ -0,0 +1,28 @@
+[{
```
Please fix the indentation

mahikaw replied:
done. 👍

mahikaw commented:
@iamareebjamal the track names were in caps in the actual sample. Does it make a very big difference in general?

iamareebjamal commented:
@mahikaw OK. Not a big issue. But please fix the formatting

iamareebjamal commented on sample/F8/sessions:
```diff
@@ -0,0 +1,1231 @@
+			"name": "1 hour session",
+			"length": "01.00"
+		},
+		"track": {},
```
Please assign full track, this will save a track with null fields in database

iamareebjamal commented on sample/F8/tracks:
```diff
@@ -0,0 +1,209 @@
+		"name": "ADS & MONETIZATION",
+		"color": "#BF532E",
+		"description": null,
+		"location": "C",
```
Similarly here. Malformed JSON

mahikaw commented:
@iamareebjamal @Shailesh351 @arp95  Please see if there are any other changes required apart from those that you have already mentioned, as I have incorporated them.

iamareebjamal commented:
@mahikaw It is still showing the same error. I tried sending PR, but there were some conflicts. I'll send it soon

mahikaw commented:
Sure thanks. Do tell if you find any other issues with this PR. @iamareebjamal :)

iamareebjamal commented:
![device-2017-06-05-173206](https://cloud.githubusercontent.com/assets/3874064/26783267/f1a9ed70-4a14-11e7-997b-1b789f81e665.png)

Missing description of sessions

iamareebjamal commented:
![device-2017-06-05-183516](https://cloud.githubusercontent.com/assets/3874064/26784967/d3c84ac8-4a1d-11e7-8704-32086b961b89.png)

There is no session for this

iamareebjamal commented:
![screenshot_20170605_184137](https://cloud.githubusercontent.com/assets/3874064/26785158/b16c6ab2-4a1e-11e7-8c61-8271f1084526.png)

Please take a look at last few sessions without track, they don't have other information too

arp95 commented on sample/F8/event:
```diff
@@ -0,0 +1,51 @@
+   "id":180,
+   "name":"F8-Facebook Developer Conference 2017",
+   "latitude":"37.329008",
+   "longitude":"-121.888794",
```
use float for latitude and longitude please.

arp95 commented on sample/F8/event:
```diff
@@ -0,0 +1,51 @@
+   "background_image":"https://drive.google.com/uc?export=download&id=0B2b0W4GqpgLFMFlkampvdGlfZVk",
+   "logo":"https://drive.google.com/uc?export=download&id=0B2b0W4GqpgLFMFlkampvdGlfZVk",
+   "organizer_name":"Facebook",
+   "organizer_description":"",
```
use the same data for description tag over here.

arp95 commented on sample/F8/microlocations:
```diff
@@ -0,0 +1,74 @@
+      "name":"Registration Desk",
+      "floor":null,
+      "latitude":"37.329008",
+      "longitude":"-121.888794",
```
same as before. please use float values.

arp95 commented on sample/F8/tracks:
```diff
@@ -0,0 +1,202 @@
+      "name":"ADS & MONETIZATION",
+      "color":"#BF532E",
+      "description":null,
+      "location":"C",
```
please check whether location is an object or array. i am not sure.

mahikaw replied:
it's just an object. :)

niranjan94 replied:
description and location can be removed from track object. Shouldn't exist and not used anywhere either.

mahikaw replied:
@niranjan94 we have an [issue](https://github.com/fossasia/open-event-android/issues/1633) about description of tracks in the android repo. Could you please confirm that we should remove track description. While location of tracks don't make sense, track description could be used I believe.

iamareebjamal replied:
@niranjan94 Those fields are still listed in the API docs, shouldn't they be removed then?

niranjan94 replied:
The fields should be removed. The current API docs are anyway deprecated due to the new API which is being built. 

We will no longer be depending on the API output for the the sample json format (since the API output will be following json api spec and is not suited for such import/export purposes). The sample will be have a separate format definition defined in the repo.

mahikaw replied:
The description too @niranjan94 ? In that case, should I close the issue mentioned above?

niranjan94 replied:
Yes please @mahikaw

arp95 commented on sample/F8/sessions:
```diff
@@ -0,0 +1,1226 @@
+      "signup_url":null,
+      "state":"accepted",
+      "timezone":"UTC",
+      "session_type":{
```
Nice! Please provide session types to other samples as well.

arp95 commented on sample/F8/sessions:
```diff
@@ -0,0 +1,1226 @@
+   {
+      "id":1,
+      "short_abstract":null,
+      "long_abstract":null,
```
No description for the session?
Please provide atleast one either short or long.

mahikaw replied:
sessions like registrations don't have any descriptions. I have provided all descriptions wherever possible.

arp95 commented on sample/F8/sessions:
```diff
@@ -0,0 +1,1226 @@
+         "name":"1 hour session",
+         "length":"01.00"
+      },
+      "track": null,
```
please provide the track to which this session belongs.

mahikaw replied:
Again, this doesn't belong to any track. :)

arp95 commented on sample/F8/sessions:
```diff
@@ -0,0 +1,1226 @@
+   },
+   {
+      "short_abstract":null,
+      "long_abstract":null,
```
same as above. please provide the description either short or long for all sessions that follow please.

mahikaw replied:
There sin't any description available for such events. :)

arp95 commented on sample/F8/sessions:
```diff
@@ -0,0 +1,1226 @@
+         "name":"2 hour session",
+         "length":"02.00"
+      },
+      "track": null,
```
same as above. please provide track for the session it belongs to all the next cases please.

mahikaw replied:
this too doesn't have any description.

arp95 commented on sample/F8/sessions:
```diff
@@ -0,0 +1,1226 @@
+         {
+            "organisation":"Facebook",
+            "id":6,
+            "name":"Justin Rogers"
```
speaker has more description over here. please check my sample. These mistakes are repeating again. please check them in all your samples.

mahikaw replied:
Yeah, I just saw the extra fields in your sample however, I didn't see it any other sample. How did you come about all of this information?

iamareebjamal replied:
API changes over time to include more attributes

mahikaw replied:
I don't have information for the new fields in my samples anyways. Shall I add them with null values? Also, in that case all the samples in the repo are outdated I believe?

iamareebjamal replied:
You should add empty strings for them instead of null, and yes, as the API changes, samples are bound to get stale

arp95 replied:
please mahika add more info for speaker here as me and areeb pointed out.

mahikaw replied:
I have added the information in speakers.json The speaker information provided in sessions is simply to link them to those sessions I believe.

niranjan94 commented on sample/F8/sessions:
```diff
@@ -0,0 +1,1226 @@
+      "id":5,
+      "subtitle":"introduction-to-the-camera-effects-platform",
+      "title":"Introduction to the Camera Effects Platform",
+      "sessionDescription":"Come hear more about Camera Effects Platform, a suite of creative tools that lets you create effects for the Facebook camera, from simple photo frames to interactive masks and effects that use Augmented Reality (AR). In this session you’ll learn what camera effects are, how to create and submit your own, and how you can connect with the audiences that matter to you by  ...
```
@mahikaw what is this new `sessionDescription` field here ? This should move to either short or long abstract.

Shailesh351 commented on sample/F8/meta:
```diff
@@ -0,0 +1,3 @@
+{
+    "root_url": "http://open-event-dev.herokuapp.com/"
```
`root_url` is url of server from where event is created.
please set it to `""` because it is not created using server

Shailesh351 commented:
There are no sessions for 19 April please add some sessions for it.
Also for location "A"

iamareebjamal commented on sample/F8/sessions:
```diff
@@ -0,0 +1,1308 @@
+         "name":"1 hour session",
+         "length":"01.00"
+      },
+      "track":"",
```
![device-2017-06-06-174904](https://cloud.githubusercontent.com/assets/3874064/26828794/98f92a8e-4ae0-11e7-8458-4143bf4dc4b0.png)

iamareebjamal commented on sample/F8/sessions:
```diff
@@ -0,0 +1,1308 @@
+         "name":"2 hour session",
+         "length":"02.00"
+      },
+      "track":"",
```
Tracks are not strings, please fix

arp95 commented on sample/F8/sessions:
```diff
@@ -0,0 +1,1308 @@
+         "name":"1 hour session",
+         "length":"01.00"
+      },
+      "track":{},
```
please provide track of this session id. if no track then remove the session.

iamareebjamal replied:
As I said before, please don't use empty tracks, use null instead

arp95 commented on sample/F8/sessions:
```diff
@@ -0,0 +1,1308 @@
+[
+   {
+      "id":1,
+      "short_abstract":"",
```
description for the event?

mahikaw replied:
@arp95 there is no description for this event. This is a registration event.

arp95 commented on sample/F8/sessions:
```diff
@@ -0,0 +1,1308 @@
+      "signup_url":"",
+      "state":"accepted",
+      "timezone":"UTC",
+      "session_type":{
```
timezone in event.json is "US/Mountain" while over here its UTC. please correct it.

arp95 commented on sample/F8/sessions:
```diff
@@ -0,0 +1,1308 @@
+         "name":"2 hour session",
+         "length":"02.00"
+      },
+      "track":{},
```
again no track.

arp95 commented on sample/F8/sessions:
```diff
@@ -0,0 +1,1308 @@
+         "length":"03.00"
+      },
+      "track":{},
+      "start_time":"2017-04-18T10:00:00",
```
please either provide tracks for session or else remove them, as it makes no sense to have a session without a track. do this for all please.

arp95 commented on sample/F8/sessions:
```diff
@@ -0,0 +1,1308 @@
+      "video":"",
+      "signup_url":"",
+      "state":"accepted",
+      "timezone":"UTC",
```
correct timezone for this and all that follow.

arp95 commented on sample/F8/sessions:
```diff
@@ -0,0 +1,1308 @@
+         "name":"F"
+      },
+      "start_time":"2017-04-18T12:30:00",
+      "comments":"",
```
wheres the "end_time" json attribute for this?

mahikaw replied:
It is there. Please have another look at it.

arp95 commented on sample/F8/sessions:
```diff
@@ -0,0 +1,1308 @@
+         "name":"ENGINEERING & OPEN SOURCE"
+      },
+      "start_time":"2017-04-18T13:30:00",
+      "speakers":[
```
"end_time" json attribute?

mahikaw replied:
end time is also there.

arp95 commented on sample/F8/sessions:
```diff
@@ -0,0 +1,1308 @@
+      },
+      "start_time":"2017-04-18T13:30:00",
+      "end_time":"2017-04-18T14:20:00",
+      "microlocation":{
```
session_type says session is of 1 hour. check the start and end time.

arp95 commented on sample/F8/sessions:
```diff
@@ -0,0 +1,1308 @@
+         "name":"G"
+      },
+      "start_time":"2017-04-18T13:30:00",
+      "end_time":"2017-04-18T14:20:00",
```
same as above. check the time.

arp95 commented on sample/F8/sessions:
```diff
@@ -0,0 +1,1308 @@
+      "id":25,
+      "title":"What can be Learned from China's Mobile Economy Boom in Past 5 Years?",
+      "subtitle":"what-can-be-learned-from-chinas-mobile-economy-boom-in-the-last-five-years",
+      "long_abstract":"There’s been 110 billion-dollar tech startups since 2010, and behind them lies the whole route of China's mobile economy boom. With a highly connected and digitalized social environment, it’s possible to fill big gaps of user demands in a more efficient and intelligent way - a pattern that can be repeated.",
```
"start_time" and "end_time" json attribute?

arp95 commented on sample/F8/sessions:
```diff
@@ -0,0 +1,1308 @@
+      "session_type":{
+         "id":1,
+         "name":"1 hour session",
+         "length":"01.00"
```
session_type doesnt match with (end_time - start_time)

arp95 commented on sample/F8/sessions:
```diff
@@ -0,0 +1,1308 @@
+      "session_type":{
+         "id":1,
+         "name":"1 hour session",
+         "length":"01.00"
```
same as above.

arp95 commented on sample/F8/sessions:
```diff
@@ -0,0 +1,1308 @@
+      "session_type":{
+         "id":1,
+         "name":"1 hour session",
+         "length":"01.00"
```
same as above.

arp95 commented on sample/F8/sessions:
```diff
@@ -0,0 +1,1308 @@
+      },
+      "start_time":"2017-04-18T16:00:00",
+      "comments":"",
+      "language":"",
```
"end_time" json attribute?

iamareebjamal commented on sample/F8/sessions:
```diff
@@ -0,0 +1,1308 @@
+      "name":"50 minutes session",
+      "length":"00.50"
+   }
+      "track":{
```
Please, always pass json through validators before posting

This is invalid

iamareebjamal commented on sample/F8/microlocations:
```diff
@@ -0,0 +1,74 @@
+   {  
+      "id":1,
+      "name":"Registration Desk",
+      "floor":"",
```
Floor is integer, not string

iamareebjamal commented:
![screenshot_20170606_202635](https://cloud.githubusercontent.com/assets/3874064/26835722/8c394c82-4af6-11e7-93f6-da52abbc428e.png)

These sessions have no information

heysadboy commented on sample/F8/speakers:
```diff
@@ -0,0 +1,2631 @@
+            "title":"Building Native Games for Facebook Gameroom"
+         }
+      ],
+      "country":"Not available",
```
I see "Not available" in countries but other fields like city, speaking experience and short biography are empty. Should we put some indicator in those too or leave them null?

mahikaw replied:
actually this I retrieved this JSON from the F8 site only. But okay, I think we can change this, yes. 👍

arp95 commented on sample/F8/sessions:
```diff
@@ -0,0 +1,1139 @@
+         "length": "00.50"
+      },
+      "track": {
+         "id": 3,
```
track with id 3 doesnt have session with id  21. please check the session.json file.

arp95 commented on sample/F8/sessions:
```diff
@@ -0,0 +1,1139 @@
+         "name": "F"
+      },
+      "start_time": "2017-04-18T14:30:00",
+      "speakers": [{
```
no "session_type" attribute? please check.

mahikaw replied:
It is there.

arp95 commented on sample/F8/sessions:
```diff
@@ -0,0 +1,1139 @@
+      },
+      "track": {
+         "id": 5,
+         "name": "VIRTUAL REALITY"
```
provide "color" attribute for all tracks in session json

arp95 commented on sample/F8/sessions:
```diff
@@ -0,0 +1,1139 @@
+      },
+      "speakers": [{
+            "organisation": "Facebook",
+            "id": 54,
```
also please add all the info about speakers here in session json itself. i think we are not referring to speaker json to get data, instead we are getting it from here.

arp95 commented on sample/F8/sessions:
```diff
@@ -0,0 +1,1139 @@
+      "long_abstract": "Interactive and fun social experiences foster greater consumer engagement in VR. This talk will focus on the APIs and features that Oculus provides to add a social layer to your app, including the Avatar SDK, VoIP, Invites, and Oculus Rooms.",
+      "end_time": "2017-04-19T14:30:00",
+      "start_time": "2017-04-19T15:30:00",
+      "comments": "",
```
"start_time" and "end_time" seem to be interchanged. please check.

Shailesh351 commented:
A : 37.329888, -121.889772         floor : 1
B : 37.329596, -121.889714         floor : 1
C : 37.329730, -121.889383         floor : 1
F : 37.329663, -121.887677         floor : 1
G : 37.329412, -121.887457        floor : 1
LL : 37.329780, -121.889613       floor : 0

@mahikaw  Update location's data :smile:

arp95 commented on sample/F8/speakers:
```diff
@@ -0,0 +1,1613 @@
+      "small":"",
+      "speaking_experience":"",
+      "sponsorship_required":"",
+      "thumbnail":""
```
please provide the images of speakers. do this for all.

mahikaw replied:
@arp95 there is no information on the official fbf8 site regarding this. i don't think it makes sense to add various types of information of the speakers if the site hasn't disclosed it.

arp95 replied:
fine.

arp95 commented on sample/F8/speakers:
```diff
@@ -0,0 +1,1613 @@
+            "title":"Litho: A Declarative Framework for Efficient UIs"
+         }
+      ],
+      "country":"",
```
no information for any speaker?
please provide the twitter or fb handles which i did on google search. see my samples

mahikaw replied:
@arp95 could you tell me more the app crashes? Where exactly did you find them?

arp95 replied:
@mahikaw please add speaker images and their short bio.
i know it takes time but we need to have a complete sample.

iamareebjamal commented on sample/F8/microlocations:
```diff
@@ -0,0 +1,58 @@
+   {  
+      "id":2,
+      "name":"LL",
+      "floor”:1,
```
Quote error again

![screenshot_20170607_172646](https://user-images.githubusercontent.com/3874064/26877148-99b162ea-4ba6-11e7-9548-37942937e212.png)

mahikaw commented:
Could be a network issue @arp95 Does it show the "download done" snack bar?

iamareebjamal commented on sample/F8/event:
```diff
@@ -0,0 +1,51 @@
+   "location_name":"San Jose Convention Center, 150 W San Carlos St  San Jose  CA USA 95113 ",
+   "start_time":"2017-04-18T10:00:00",
+   "end_time":"2017-04-19T17:00:00",
+   "timezone”:”US/Pacific”,
```
Quote issue again @mahikaw 
Please be careful with quotes, this happens way too much

arp95 commented:
@mahikaw please test the app before posting the code.

mahikaw commented:
@mariobehling @niranjan94 @arp95 @heysadboy this is the [temporary deployment](https://peaceful-earth-19050.herokuapp.com/live/preview/mahika.wason@gmail.com/F8-Facebook%20Developer%20Conference%202017/index.html). Please suggest changes if any in the JSON by looking at the web app so that I can update it. Otherwise I will go ahead with the deployment on GitHub pages.

mahikaw commented:
https://mahikaw.github.io/eventSiteWebApp/schedule.html
Is the corresponding GitHub page deployment. Please review this too @mariobehling @niranjan94

arp95 commented on sample/F8/microlocations:
```diff
@@ -0,0 +1,57 @@
+	{
+		"id": 6,
+		"name": "F",
+		"floor": 1,
```
Please provide descriptive names for locations.

mahikaw replied:
These are the locations mentioned on the official site. How can I change them.

arp95 replied:
fine no problem.

arp95 commented on sample/F8/sessions:
```diff
@@ -0,0 +1,1470 @@
+      "track":{
+         "id":2,
+         "name":"CAMERA",
+         "color":"#FF3046"
```
font color here too.

mahikaw replied:
There is no need for that. It works just by adding them in tracks.json

arp95 commented on sample/F8/sessions:
```diff
@@ -0,0 +1,1470 @@
+            "long_biography":"",
+            "small":"",
+            "photo":"",
+            "thumbnail":"",
```
add speaker images. theres no point of an event without the images. also provide relevant info of speaekers.

mahikaw replied:
There is no image or more information mentioned on the site. If fbf8 speakers chose to give this amount of information, I doubt if I should search the net for their images and information.

arp95 commented on sample/F8/sponsors:
```diff
@@ -0,0 +1 @@
+[]
```
mahika i know there arent any sponsers but still can you add fb as a sponser.
i was made to add mozilla for my mozilla event.

arp95 replied:
add sponsor as i said.

arp95 commented on sample/F8/event:
```diff
@@ -0,0 +1,51 @@
+	"organizer_description": "Join us for our annual 2-day event, where developers and businesses come together to explore the future of technology. Learn how Facebook connects the world through new products and innovation. This year’s event is bigger than ever – with more than 50 sessions, interactive experiences, and the opportunity to meet one-on-one with members of the Facebook team.",
+	"event_url": "https://www.fbf8.com",
+	"social_links": [
+
```
add twitter social link for web app please.

arp95 commented:
@Princu7 please review it so that it can merged now.

Princu7 commented:
Most of the points have already been mentioned. Add Copyright and sponsors and social links especially Twitter to the sample.
@mariobehling @mahikaw @arp95 
Please clarify one more thing. If we are adding images, why aren't we providing a speaker image already in the sample rather than a link to the Google cloud storage.  A sample zip should be complete containing all the information inside it. Please correct if I am wrong. Thanks!!

arp95 commented:
@Princu7
in the android app we are having only two parts
1. view speaker images through links
2. view speaker images through local ones.

so right now we were referring to links to provide images as there was no support for local speaker images till now.
will be adding local images then. just to ensure web app takes images through local one?

Princu7 commented:
@arp95 Oh. Okay. Thanks for explaining. No webapp supports both modes: local and internet. Just wanted to confirm why we are including links to the speaker in the zip file because on the earlier zips eg: FOSSASIA 16 and 15,  the images of the speakers were there in the zip file itself.

And please let's wait for other suggestions before adding local images to the zip file.

arp95 commented:
@Princu7 if web app supports internet ones then are you considering only the ones from eventyay?
Also should i include the images locally to make it work? Please confirm.

Princu7 commented:
@arp95 So, I tried to find out the cause where the error is coming from. Here is the explanation:-

The problem is that in the web app, we don't directly render the image from the source. We download it from there and save it onto a folder named `images` and then specify this local path in the `src` field of the image tag.

The source of the file is something like this:
`https://drive.google.com/uc?export=download&id=0B6k_UTO5VOZKc18xTTJBSnlsTjQ`

The name of the file being saved is of this type:
`uc?export=download&id=0B6k_UTO5VOZKc18xTTJBSnlsTjQ` 

and the corresponding src specified will be
`images/speakers/uc?export=download&id=0B6k_UTO5VOZKc18xTTJBSnlsTjQ`

This is the major thing which is causing the problem. There are other issues but they are easy to fix. Since the `src` field contains a `?` in between which generally means the start of query parameters and end of normal URL, the browser starts searching for an image with the src `images/speakers/uc` which it is not able to find and hence the image is not rendered at the end.

So, even if we are adding images locally, the file names should be valid and shouldn't contain a `?` in between them. I am trying to fix the error. Thanks!!

mariobehling commented:
> A sample zip should be complete containing all the information inside it. 

Yes, exactly. This is the point. All data must be in the sample. I don't see the what is the question. Follow the FOSSASIA sample and include all images in the sample.

mariobehling commented:
> The source of the file is something like this: https://drive.google.com/uc?export=download&id=0B6k_UTO5VOZKc18xTTJBSnlsTjQ

Don't store images on Google Drive! Not sure, how to make it clear. Why should we make our samples depend on Google Drive. I am unclear where you got the idea to store images on Google Drive.

Follow the sample FOSSASIA app. Include images in the sample itself!

mahikaw commented:
@mariobehling The speakers have minimal information in the [official site](https://www.fbf8.com) which excludes their description, images and social links. Other than that I have un-capitalized the track names, added copyright, added the available social links of the event and changed the room names to "Hall A" and so on, as suggested. Please review the PR now. @Princu7 @arp95

arp95 commented on sample/F8/sessions:
```diff
@@ -0,0 +1,1470 @@
+            "icon":"",
+            "long_biography":"",
+            "small":"",
+            "photo":"",
```
add speaker images. if not available on site please google them. do this for all

arp95 commented on sample/F8/speakers:
```diff
@@ -0,0 +1,1552 @@
+      "facebook":"",
+      "github":"",
+      "linkedin":"",
+      "featured":false,
```
remember to add atleast 10-15 featured speakers. do that by making "featured" field as true.

iamareebjamal commented on sample/F8/event:
```diff
@@ -0,0 +1,54 @@
+	"end_time": "2017-04-19T17:00:00",
+	"timezone": "US / Pacific",
+	"description": "Join us for our annual 2-day event, where developers and businesses come together to explore the future of technology. Learn how Facebook connects the world through new products and innovation. This year’s event is bigger than ever – with more than 50 sessions, interactive experiences, and the opportunity to meet one-on-one with members of the Facebook team.",
+	"background_image": "/imagesbackground.png",
```
I don't seem to find any `/imagesbackground.png`, isn't this supposed to be `/images/background.png`

@arp95 This is the reason generator is crashing for this zip

@mahikaw and @arp95 Please test the zips locally and check that the apps are working, both web and Android

If not, raise the issue, but first try to locally fix the issue, don't delegate it on `the generator is not working correctly`

iamareebjamal replied:
And please align with what has been done on the app. I am not usually bothered by checking issues here and there, but today I spent hours cross-checking every zip to see how a feature that was implemented not working now and is the generator really not working and giving errors on all zips. Both of which were false. I usually feel responsible for these kinds of things because I made the last changes in the PR

arp95 replied:
@iamareebjamal again i come to my point.
web app gnerator gives the correct version but the android one isnt. this again brings us to the point where two domains are working differently. If the web app is able to generate the app and handle such mistakes why not the app generator of android part.
anyways having a look

arp95 replied:
@iamareebjamal i am not blaming you at all.
but if i am adding 200+ sessions a small mistake can happen. so help might be needed in such cases because you get the web app but not the android app.

iamareebjamal replied:
You don't seem to understand:
1. Web App generator does not need to resize images
2. There is no `database` or `cached` version of files needed for the web app

If the sample is failing, it means it fails either on Android App generator or the web app generator.
They will always work differently in some way or another, the point is that an issue must only be raised when it is actually one, not in the case there's any error in the zips

arp95 replied:
so if i am naming the sample images properly.
does it mean there is a problem resizing them in the android part

iamareebjamal replied:
I'm looking at the issue, most probably there is issue about the wrong file type. We should generate ic_launcher.png and the one being generated is jpg, meaning a JPEG file is being given as logo, which is incorrect and causing the error

iamareebjamal commented on sample/F8/event:
```diff
@@ -0,0 +1,54 @@
+	"end_time": "2017-04-19T17:00:00",
+	"timezone": "US / Pacific",
+	"description": "Join us for our annual 2-day event, where developers and businesses come together to explore the future of technology. Learn how Facebook connects the world through new products and innovation. This year’s event is bigger than ever – with more than 50 sessions, interactive experiences, and the opportunity to meet one-on-one with members of the Facebook team.",
+	"background_image": "/images/background.png",
```
@mahikaw You did not update the zip. Now someone might think that the JSON is correct but generator is crashing so there must be a problem in generator

mahikaw replied:
@iamareebjamal yes, I am working on it right now. I will update it only after I finalise my work. Please try and understand that there can be trivial errors in the samples as they are being done manually. No one is suggesting anything about the generator, we are all just trying to complete the samples collaboratively.

arp95 commented on sample/F8/event:
```diff
@@ -0,0 +1,54 @@
+	"description": "Join us for our annual 2-day event, where developers and businesses come together to explore the future of technology. Learn how Facebook connects the world through new products and innovation. This year’s event is bigger than ever – with more than 50 sessions, interactive experiences, and the opportunity to meet one-on-one with members of the Facebook team.",
+	"background_image": "/images/background.png",
+	"logo": "https://drive.google.com/uc?export=download&id=0B2b0W4GqpgLFMFlkampvdGlfZVk",
+	"organizer_name": "Facebook",
```
use local images please.

arp95 commented on sample/F8/tracks:
```diff
@@ -0,0 +1,195 @@
+      "track_image_url": "",
+      "font_color": "white"
+   },
+   {
```
make "font_color" black for light backgrounds as @Princu7 pointed out.

arp95 commented on sample/F8/sessions:
```diff
@@ -0,0 +1,1470 @@
+            "icon":"",
+            "long_biography":"",
+            "small":"",
+            "photo":"",
```
add local images please.

arp95 commented on sample/F8/sessions:
```diff
@@ -0,0 +1,1470 @@
+         },
+         {
+            "organisation":"Facebook",
+            "id":32,
```
please use same speaker info. one has more attributes than the other. please use the same format for each speaker and find images for speakers on the net.

niranjan94 commented on sample/F8/tracks:
```diff
@@ -0,0 +1,195 @@
+         }
+      ],
+      "track_image_url": "",
+      "font_color": "white"
```
Use hexcode please. For uniformity.

mahikaw commented:
Done, please check now.

mahikaw commented:
@iamareebjamal @arp95 @Princu7 Please review this sample.

iamareebjamal commented on sample/F8/sponsors:
```diff
@@ -0,0 +1,11 @@
+         "sponsor_type": "Diamond Sponsor", 
+         "url": "www.facebook.com", 
+         "description": ""
+     }, 
```
Invalid JSON, please fix

iamareebjamal replied:
Please set up a gulp/grunt file to pass all json through validators before committing. Thanks :smile: 

This will make it easier for both you and us

iamareebjamal commented:
@mahikaw You did not update the zip

mahikaw commented:
@iamareebjamal Yes, I am aware and I'll be updating it soon and I will inform you when I do. Thanks for pointing it out. 👍

mahikaw commented:
@iamareebjamal @arp95 Please review now.

iamareebjamal commented:
Furthermore, most of facebook employees have (fortunately) facebook accounts, so it'll be really easy for you to find their images. A quick google search even showed me that most of them had public photos available directly from google

iamareebjamal commented:
@mahikaw @arp95 Try this

```bash
sudo -H pip3 install coala-bears  
coala --files * --bears JSONFormatBear
```

Not only tells if the json is invalid, also formats and prompts before applying the patch, showing the changes it'll do. Is very configurable too. Furthermore, you can create a little bash script to run this and recreate the zip in one go

mahikaw commented:
@iamareebjamal google search shows a lot of images for sure, but I can't just put them against a speaker's name without any certainty of who exactly it is in the image. Also, as I have repeatedly mentioned- the F8 site didn't reveal any speaker images, I don't think we should go about googling them and getting wrong data in the sample(There is a high probability of misinformation here).


Also, thanks for introducing the coals-bear thing, does seem useful 👍 ,  but I request you to post such things to only the public channel as this forum is for things related to this particular issue. 😄

iamareebjamal commented:
Some of the speakers are very well known and not bound to be mistaken. Anyways, that's my opinion

mahikaw commented:
@iamareebjamal Could you please point out the popular ones to me then? I will add the images.

iamareebjamal commented:
@mahikaw Sure. Till then, please take a look at missing sessions

mariobehling commented:
@mahikaw Thank you. Please also look into the following:

* Add twitter https://twitter.com/fbplatform
* Add Instagram https://www.instagram.com/explore/tags/fbf8/
* Add GitHub https://github.com/Facebook
* License of event is Copyright and not Public Domain as far as I can see.
* Please research and add pictures of speakers
* Add position of speakers (Looks like all speakers from Facebook, is this correct?)

We also need to show our video capabilities in the apps.
Please add videos from here: https://developers.facebook.com/videos

arp95 commented on sample/F8/speakers:
```diff
@@ -0,0 +1,3195 @@
+      "twitter":"",
+      "facebook":"",
+      "github":"",
+      "linkedin":"",
```
please surf for the links. provide for quality info.

arp95 commented on sample/F8/speakers:
```diff
@@ -0,0 +1,3195 @@
+      "city":"",
+      "gender":"male",
+      "heard_from":"",
+      "icon":"",
```
i dont see any featured speakers??
please provide some around 10-15.

mariobehling commented:
@mahikaw As partners please also add the logos of Facebook companies:

* Facebook Payments Inc. (https://www.facebook.com/payments_terms/privacy)
* Atlas (http://atlassolutions.com/privacy-policy)
* Instagram LLC (http://instagram.com/about/legal/privacy/)
* Onavo (http://www.onavo.com/privacy_policy)
* Moves (http://moves-app.com/privacy)
* Oculus (http://www.oculus.com/privacy/)
* WhatsApp Inc. (http://www.whatsapp.com/legal/#Privacy)
* Masquerade (https://www.facebook.com/msqrd/privacy)
* CrowdTangle (https://www.crowdtangle.com/privacy)
